packages/manager/apps/pci-instances/src/pages/instances/datagrid/hooks/useDatagridOperationsPolling.ts

```diff
@@ -1,70 +1,88 @@
-import { useCallback, useState } from 'react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
 import { useProjectId } from '@/hooks/project/useProjectId';
 import { TOperation } from '@/types/operation/entity.type';
-import { useOperationsPolling } from '@/data/hooks/instance/polling/useOperationsPolling';
+import { useOperationsPolling } from '@/data/hooks/operation/polling/useOperationsPolling';
 import { shouldRetryAfterNot404Error } from '@/data/hooks/instance/polling/useInstancesPolling';
+import {
+  isInstanceCreationOperationInProgress,
+  isInstanceReinstallOperationInProgress,
+  isSubOperationCopyImage,
+  isSubOperationCopyImageInProgress,
+} from '@/utils/operation/operations.utils';
 
-const isInstanceCreationOperationInProgress = (operation: TOperation) =>
-  operation.section === 'instance' &&
-  operation.action === 'create' &&
-  operation.status === 'in-progress';
-
-const isInstanceReinstallOperationInProgress = (operation: TOperation) =>
-  operation.section === 'instance' &&
-  operation.action === 'reinstall' &&
-  operation.status === 'in-progress';
+const getNumberOfOperationsWithImageCopyInProgress = (
+  operations: TOperation[] = [],
+) =>
+  operations
+    .filter(
+      (operation) =>
+        isInstanceCreationOperationInProgress(operation) ||
+        isInstanceReinstallOperationInProgress(operation),
+    )
+    .flatMap((operation) => operation.subOperations)
+    .filter(isSubOperationCopyImageInProgress).length;
 
-const isSubOperationCopyImageInProgress = (
-  subOperation?: Required<TOperation>['subOperations'][number],
+const getNumberOfCreationOperationWithoutImageCopy = (
+  operations: TOperation[] = [],
 ) =>
-  subOperation &&
-  subOperation.section === 'image' &&
-  subOperation.action === 'copytoregion' &&
-  subOperation.status === 'in-progress';
+  operations
+    .filter(isInstanceCreationOperationInProgress)
+    .filter(
+      (operation) => !operation.subOperations?.some(isSubOperationCopyImage),
+    ).length;
+
+const getNumberOfOperationInProgress = (operations: TOperation[] = []) =>
+  getNumberOfOperationsWithImageCopyInProgress(operations) +
+  getNumberOfCreationOperationWithoutImageCopy(operations);
 
 export const useDatagridOperationsPolling = (onComplete?: () => void) => {
   const projectId = useProjectId();
 
   const [
-    numberOfDistantBackupCopyRunning,
-    setNumberOfDistantBackupCopyRunning,
+    numberOfOperationsInProgress,
+    setNumberOfOperationsInProgress,
   ] = useState(0);
 
   const onSuccess = useCallback(
     (operations?: TOperation[]) => {
       if (!operations) return;
```

Comment:
You should also eject `if operations.length === 0` to prevent from calling unnecessary functions